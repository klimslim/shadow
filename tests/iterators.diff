001- error_reporting_at_start: 12345
001+ error_reporting_at_start: 30719
     inst_accounts_file
     inst_calls_file
     inst_cases_file
--
     inst_project_file
     inst_purchasedlineitems_file
     inst_tasks_file
014- string(%d) "Executed content of %s (up to %d bytes)"
015- string(%d) "Executed content of %s (up to %d bytes)"
014+ string(28) "<?php echo "instance rest";
015+ "
016+ string(28) "<?php echo "template rest";
017+ "
