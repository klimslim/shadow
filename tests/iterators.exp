error_reporting_at_start: 12345
inst_accounts_file
inst_calls_file
inst_cases_file
inst_contacts_file
inst_documents_file
inst_externalusers_file
inst_meetings_file
inst_messages_file
inst_notes_file
inst_project_file
inst_purchasedlineitems_file
inst_tasks_file
string(%d) "Executed content of %s (up to %d bytes)"
string(%d) "Executed content of %s (up to %d bytes)"