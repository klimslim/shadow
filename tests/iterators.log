
---- EXPECTED OUTPUT
error_reporting_at_start: 12345
inst_accounts_file
inst_calls_file
inst_cases_file
inst_contacts_file
inst_documents_file
inst_externalusers_file
inst_meetings_file
inst_messages_file
inst_notes_file
inst_project_file
inst_purchasedlineitems_file
inst_tasks_file
string(%d) "Executed content of %s (up to %d bytes)"
string(%d) "Executed content of %s (up to %d bytes)"
---- ACTUAL OUTPUT
error_reporting_at_start: 30719
inst_accounts_file
inst_calls_file
inst_cases_file
inst_contacts_file
inst_documents_file
inst_externalusers_file
inst_meetings_file
inst_messages_file
inst_notes_file
inst_project_file
inst_purchasedlineitems_file
inst_tasks_file
string(28) "<?php echo "instance rest";
"
string(28) "<?php echo "template rest";
"
---- FAILED
