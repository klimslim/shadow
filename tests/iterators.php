<?php
echo "error_reporting_at_start: " . ini_get('error_reporting') . "\n";

$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Accounts";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Calls";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Cases";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Contacts";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Documents";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/ExternalUsers";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Meetings";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Messages";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Notes";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Project";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/PurchasedLineItems";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Tasks";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/api/rest.php";
$fp = fopen($path, "r");
var_dump(fread($fp, 1024));
fclose($fp);
$path = "tests/fixtures/templatedir/api/rest.php";
$fp = fopen($path, "r");
var_dump(fread($fp, 1024));
fclose($fp);
?>
