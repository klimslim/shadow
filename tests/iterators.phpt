--TEST--
Check iterators
--SKIPIF--
<?php if (!extension_loaded("shadow")) print "skip"; ?>
--INI--
shadow.enabled=1
shadow.override_all=1
shadow.mkdir_mask=0755
shadow.debug=4096
shadow.cache_size=0
shadow.instance_path=tests/fixtures/instance
shadow.template_path=tests/fixtures/templatedir
--FILE--
<?php
echo "error_reporting_at_start: " . ini_get('error_reporting') . "\n";

$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Accounts";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Calls";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Cases";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Contacts";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Documents";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/ExternalUsers";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Meetings";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Messages";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Notes";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Project";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/PurchasedLineItems";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/qwe/modulebuilder/builds/qweqwe/SugarModules/modules/Tasks";
$dit = new DirectoryIterator($path);
foreach ($dit as $f) {
	if($f->isDot()) continue;
	echo $f->getFilename()."\n";
}
$path = "tests/fixtures/instance/api/rest.php";
$fp = fopen($path, "r");
var_dump(fread($fp, 1024));
fclose($fp);
$path = "tests/fixtures/templatedir/api/rest.php";
$fp = fopen($path, "r");
var_dump(fread($fp, 1024));
fclose($fp);
?>
--EXPECTF--
error_reporting_at_start: 12345
inst_accounts_file
inst_calls_file
inst_cases_file
inst_contacts_file
inst_documents_file
inst_externalusers_file
inst_meetings_file
inst_messages_file
inst_notes_file
inst_project_file
inst_purchasedlineitems_file
inst_tasks_file
string(%d) "Executed content of %s (up to %d bytes)"
string(%d) "Executed content of %s (up to %d bytes)"
