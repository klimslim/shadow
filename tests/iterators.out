error_reporting_at_start: 30719
inst_accounts_file
inst_calls_file
inst_cases_file
inst_contacts_file
inst_documents_file
inst_externalusers_file
inst_meetings_file
inst_messages_file
inst_notes_file
inst_project_file
inst_purchasedlineitems_file
inst_tasks_file
string(28) "<?php echo "instance rest";
"
string(28) "<?php echo "template rest";
"